r/src/RecordBatch.cpp

```diff
@@ -0,0 +1,145 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow_types.h"
+#include <arrow/io/file.h>
+#include <arrow/ipc/writer.h>
+#include <arrow/ipc/reader.h>
```

Comment:
May be more compatible to use `arrow/io/api.h` and `arrow/ipc/api.h` here. That's what we do in Python https://github.com/apache/arrow/blob/master/python/pyarrow/includes/libarrow.pxd#L758